src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -0,0 +1,594 @@
+
+    // Get the best EGL config matching the requested video settings
+    m_config = getBestConfig(m_display, VideoMode::getDesktopMode().bitsPerPixel, settings);
+    updateSettings();
+
+    // Create EGL context
+    createContext(shared);
+    createSurface(width, height, VideoMode::getDesktopMode().bitsPerPixel, false);
+}
+
+
+////////////////////////////////////////////////////////////
+DRMContext::~DRMContext()
+{
+    // Deactivate the current context
+    EGLContext currentContext = eglCheck(eglGetCurrentContext());
+
+    if (currentContext == m_context)
+    {
+        eglCheck(eglMakeCurrent(m_display, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT));
+    }
+
+    // Destroy context
+    if (m_context != EGL_NO_CONTEXT)
+    {
+        eglCheck(eglDestroyContext(m_display, m_context));
+        m_context = EGL_NO_CONTEXT;
+    }
+
+    // Destroy surface
+    if (m_surface != EGL_NO_SURFACE)
+    {
+        eglCheck(eglDestroySurface(m_display, m_surface));
+        m_surface = EGL_NO_SURFACE;
+    }
+
+    if (m_currentBO)
+        gbm_surface_release_buffer(m_gbmSurface, m_currentBO);
+
+    if (m_nextBO)
+        gbm_surface_release_buffer(m_gbmSurface, m_nextBO);
+
+    if (m_gbmSurface)
+        gbm_surface_destroy(m_gbmSurface);
+
+    contextCount--;
+    if (contextCount == 0)
+        cleanup();
+}
+
+
+////////////////////////////////////////////////////////////
+bool DRMContext::makeCurrent(bool current)
+{
+    const EGLSurface surface = current ? m_surface : EGL_NO_SURFACE;
+    const EGLContext context = current ? m_context : EGL_NO_CONTEXT;
+    return m_surface != EGL_NO_SURFACE && eglMakeCurrent(m_display, surface, surface, context);
+}
+
+
+////////////////////////////////////////////////////////////
+void DRMContext::display()
+{
+    if (m_surface == EGL_NO_SURFACE)
+        return;
+
+    if (!m_scanOut)
+    {
+        eglCheck(eglSwapBuffers(m_display, m_surface));
+        return;
+    }
+
+    // Handle display of buffer to the screen
+    drm_fb* fb = NULL;
+
+    if (!waitForFlip(-1))
+        return;
+
+    if (m_currentBO)
+    {
+        gbm_surface_release_buffer(m_gbmSurface, m_currentBO);
+        m_currentBO = NULL;
+    }
+
+    eglCheck(eglSwapBuffers(m_display, m_surface));
+
+    m_currentBO = m_nextBO;
+
+    // This call must be preceeded by a single call to eglSwapBuffers()
+    m_nextBO = gbm_surface_lock_front_buffer(m_gbmSurface);
+
+    if (!m_nextBO)
+        return;
+
+    fb = drm_fb_get_from_bo(m_nextBO);
+    if (!fb)
+    {
+        err() << "Failed to get FB from buffer object" << std::endl;
+        return;
+    }
+
+    //
+    // If first time, need to first call drmModeSetCrtc()
+    //
+    if (!m_shown)
+    {
+        if (drmModeSetCrtc(drmNode.fd, drmNode.crtc_id, fb->fb_id, 0, 0,
+            &drmNode.connector_id, 1, drmNode.mode))
+        {
+            err() << "Failed to set mode: " << std::strerror(errno) << std::endl;
+            std::abort();
```

Comment:
No abort here either

Reply:
[            std::abort();](https://github.com/SFML/SFML/blob/f7c88ee7ef4e1c705531cd614efb7dcff1f873cb/src/SFML/Window/Unix/Display.cpp#L69)
What do you propose? In Unix/Display.cpp I also found the use of `abort()` btw.

Reply:
I guess in that case, we can keep it for now.